### PR TITLE
remove IAM old migration code

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
remove IAM old migration code

## Motivation and Context
```
commit 7bdaf9bc50eb49061e8a71b9cf471eae5db654ff
Author: Aditya Manthramurthy <donatello@users.noreply.github.com>
Date:   Wed Jul 24 17:34:23 2019 -0700

    Update on-disk storage format for users system (#7949)
```

Bonus: fixes a bug when etcd keys were being re-encrypted.

## How to test this PR?
Nothing to test this is a breaking change

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/config-encrypted.go [hunk 1]
```diff
@@ -146,44 +153,54 @@ func migrateConfigPrefixToEncrypted(objAPI ObjectLayer, encrypted bool) error {
...
-			if GlobalKMS != nil {
-				data, err = config.EncryptBytes(GlobalKMS, data, kms.Context{
-					obj.Bucket: path.Join(obj.Bucket, obj.Name),
-				})
-				if err != nil {
-					return err
+		if !utf8.Valid(data) {
+			pdata, err := madmin.DecryptData(globalActiveCred.String(), bytes.NewReader(data))
+			if err != nil {
+				if GlobalKMS != nil {
```
Comment (donatello) [inline cmd/config-encrypted.go:172]: Seems to missing the case where `err != nil && GlobalKMS == nil`? This condition appears to be not-handled in `migrateIAMConfigsEtcdToEncrypted` too.

**Review by donatello — CHANGES_REQUESTED** [review]

File: cmd/config-encrypted.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/config-encrypted.go:172]: well there is nothing we can do at that point if content cannot be decrypted @donatello - so its not handled. what are you thinking that we should do?

Comment (donatello) [reply cmd/config-encrypted.go:172]: In the other case when decryption fails, we do `return fmt.Errorf("Decrypting IAM config failed %w, possibly credentials are incorrect", err)` - we can do the same here?

Comment (donatello) [reply cmd/config-encrypted.go:172]: Otherwise looks like we are saving/re-encrypting some garbage output.

Comment (harshavardhana) [reply cmd/config-encrypted.go:172]: Let me add that perhaps this has been a cause of multiple issues in the past.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by donatello — APPROVED** [review]

**Review by krisis — APPROVED** [review]

**Pull request merged**